packages/flutter/test/widgets/image_test.dart

```diff
@@ -2123,6 +2123,57 @@ void main() {
 
     codec.dispose();
   });
+
+  testWidgets('Do not clean up rendered images if the Image Provider is not updated', (WidgetTester tester) async {
```

Comment:
```suggestion
  testWidgets('Do not clear rendered images if the image provider is not updated', (WidgetTester tester) async {
```

Reply:
Also this test doesn't seem to work. I've removed your changes (basically reverted `if (updatedImage && !widget.gaplessPlayback)`) and the test still passes.

Reply:
This PR does solve the problem I'm having, but it needs more rigorous testing.

Did not solve this problem.
> Image seems to be missing a builder, between loadingBuilder and frameBuilder.